<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>800</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Edge Detection</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0" rowspan="2">
       <widget class="QGroupBox" name="groupBox">
        <property name="minimumSize">
         <size>
          <width>700</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>700</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string/>
        </property>
        <widget class="QSlider" name="upperSlider">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>500</y>
           <width>641</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimum">
          <number>50</number>
         </property>
         <property name="maximum">
          <number>120</number>
         </property>
         <property name="value">
          <number>70</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="upperLabel">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>460</y>
           <width>221</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Upper Threshold</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="sobelButton">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>110</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Sobel</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="prewittButton">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>60</y>
           <width>110</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Prewitt</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="laplacianButton">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>90</y>
           <width>110</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Laplacian</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="cannyButton">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>120</y>
           <width>110</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Canny</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="dericheButton">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>150</y>
           <width>110</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Deriche</string>
         </property>
        </widget>
        <widget class="QLabel" name="alphaLabel">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>380</y>
           <width>221</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Alpha</string>
         </property>
        </widget>
        <widget class="QSlider" name="alphaSlider">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>420</y>
           <width>641</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>5</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="lowerLabel">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>540</y>
           <width>221</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Lower Threshold</string>
         </property>
        </widget>
        <widget class="QSlider" name="lowerSlider">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>580</y>
           <width>641</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>60</number>
         </property>
         <property name="value">
          <number>35</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QPushButton" name="saveButton">
         <property name="geometry">
          <rect>
           <x>562</x>
           <y>20</y>
           <width>121</width>
           <height>51</height>
          </rect>
         </property>
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
        <widget class="QPushButton" name="processButton">
         <property name="geometry">
          <rect>
           <x>290</x>
           <y>630</y>
           <width>371</width>
           <height>111</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">font: 700 14pt &quot;Arial&quot;;</string>
         </property>
         <property name="text">
          <string>PROCESS IMAGE</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="upperSpinBox">
         <property name="geometry">
          <rect>
           <x>420</x>
           <y>460</y>
           <width>50</width>
           <height>25</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="maximum">
          <number>120</number>
         </property>
         <property name="value">
          <number>70</number>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="alphaSpinBox">
         <property name="geometry">
          <rect>
           <x>420</x>
           <y>390</y>
           <width>61</width>
           <height>36</height>
          </rect>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
        <widget class="QSpinBox" name="lowerSpinBox">
         <property name="geometry">
          <rect>
           <x>420</x>
           <y>540</y>
           <width>50</width>
           <height>25</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="maximum">
          <number>60</number>
         </property>
         <property name="value">
          <number>35</number>
         </property>
        </widget>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="processedImageLabel">
        <property name="maximumSize">
         <size>
          <width>1012</width>
          <height>490</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="originalImageLabel">
        <property name="maximumSize">
         <size>
          <width>1012</width>
          <height>490</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1500</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionClose"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>upperSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>upperSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>382</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lowerSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lowerSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>345</x>
     <y>615</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>585</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
